### PR TITLE
[RTM] Add editInformation to EditMask to cancel saving from PrePersist

### PR DESCRIPTION
**Pull request by zonky2**
Add editInformation to EditMask to cancel saving from PrePersist

### CODE REVIEW

File: src/Contao/View/Contao2BackendView/EditMask.php [hunk 1]
```diff
@@ -642,16 +642,21 @@ protected function getHeadline()
      */
     protected function doPersist()
     {
-        $environment   = $this->getEnvironment();
-        $dataProvider  = $environment->getDataProvider($this->model->getProviderName());
-        $inputProvider = $environment->getInputProvider();
+        $environment     = $this->getEnvironment();
+        $dataProvider    = $environment->getDataProvider($this->model->getProviderName());
+        $inputProvider   = $environment->getInputProvider();
+        $editInformation = System::getContainer()->get('cca.dc-general.edit-information');
```
Comment (dmolineus) [inline src/Contao/View/Contao2BackendView/EditMask.php:648]: Please use dependency injection

**Review by dmolineus — CHANGES_REQUESTED** [review]

File: src/Contao/View/Contao2BackendView/EditMask.php (hunk 1 above)
Comment (discordier) [reply src/Contao/View/Contao2BackendView/EditMask.php:648]: @dmolineus I am unsure how we should handle this.
I mean, the current code is already using the container directly, however only in a narrow context.
Now we would end up using it always for every save operation.

I tend towards using DI here, but we would still need the fallback non the less.

@zonky2 If you want to change it in 2.3, why not already now when you are already altering the code. Where is the reasoning to revisit the exact same code twice?

File: src/Contao/View/Contao2BackendView/EditMask.php [hunk 2]
```diff
@@ -642,16 +642,21 @@ protected function getHeadline()
...
 
         if (!$this->model->getMeta(ModelInterface::IS_CHANGED)) {
             return true;
         }
 
         $this->handlePrePersist();
 
+        if ($editInformation->hasAnyModelError()) {
```
Comment (discordier) [inline src/Contao/View/Contao2BackendView/EditMask.php:656]:
```suggestion
        if ($editInformation->getModelError($this->model)) {
```
You are only interested in the errors within the current model and not all models being contained in the system, aren't you?

File: src/Contao/View/Contao2BackendView/EditMask.php (hunk 1 above)
Comment (discordier) [reply src/Contao/View/Contao2BackendView/EditMask.php:648]: On the other hand, this stateful service still feels so wrong to me.

**Review by discordier — CHANGES_REQUESTED** [review]

File: src/Contao/View/Contao2BackendView/EditMask.php (hunk 2 above)
Comment (zonky2) [reply src/Contao/View/Contao2BackendView/EditMask.php:656]: you could be right ...

File: src/Contao/View/Contao2BackendView/EditMask.php (hunk 1 above)
Comment (zonky2) [reply src/Contao/View/Contao2BackendView/EditMask.php]: fixed

File: src/Contao/View/Contao2BackendView/EditMask.php (hunk 2 above)
Comment (zonky2) [reply src/Contao/View/Contao2BackendView/EditMask.php]: fixed

**Pull request merged**